tools/http-test-helper/src/main/java/org/enso/shttp/cloud_mock/PathResolver.java

```diff
@@ -0,0 +1,69 @@
+package org.enso.shttp.cloud_mock;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+public class PathResolver implements CloudHandler {
+
+  private static final String PATH_RESOLVE = "path/resolve";
+
+  private final String ORGANIZATION_NAME = "Test.ORG";
+
+  private final AssetStore assetStore;
+  private final ObjectMapper jsonMapper = new ObjectMapper();
+
+  public PathResolver(AssetStore assetStore) {
+    this.assetStore = assetStore;
+  }
+
+  @Override
+  public boolean canHandle(String subPath) {
+    return subPath.startsWith(PATH_RESOLVE);
+  }
+
+  private final Pattern pathPattern = Pattern.compile("enso://(.+?)/(.+)");
+
+  @Override
+  public void handleCloudAPI(CloudExchange exchange) throws IOException {
+    JsonNode root = jsonMapper.readTree(exchange.decodeBodyAsText());
+    String path = root.get("path").asText();
+    var matcher = pathPattern.matcher(path);
+    if (!matcher.matches()) {
+      exchange.sendResponse(400, "Invalid path: " + path);
+      return;
+    }
+
+    String organization = matcher.group(1);
+    String subPath = matcher.group(2);
+
+    if (!organization.equals(ORGANIZATION_NAME)) {
+      exchange.sendResponse(404, "Organization not found: " + organization);
+      return;
+    }
+
+    // The latter condition is a workaround for https://github.com/enso-org/cloud-v2/issues/1173 and it may be removed once that is fixed
+    boolean isRoot = subPath.isEmpty() || subPath.equals("/");
```

Comment:
TODO this workaround should probably also be removed once https://github.com/enso-org/cloud-v2/issues/1173 is fixed, as then it will no longer be needed

```suggestion
    boolean isRoot = subPath.isEmpty();
```